apps/intellij-plugin/src/main/kotlin/ai/hash/rustextra/projectView/ModuleDirectoryNode.kt

```diff
@@ -0,0 +1,36 @@
+package ai.hash.rustextra.projectView
+
+import com.intellij.ide.projectView.NodeSortOrder
+import com.intellij.ide.projectView.NodeSortSettings
+import com.intellij.ide.projectView.ViewSettings
+import com.intellij.ide.projectView.impl.nodes.PsiDirectoryNode
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiDirectory
+
+class ModuleDirectoryNode(project: Project?, value: PsiDirectory, viewSettings: ViewSettings?): PsiDirectoryNode(project, value, viewSettings) {
+    // This is needed, as otherwise the `GroupByTypeComparator` won't fall back to the `AlphaComparator`
+    override fun getTypeSortWeight(sortByType: Boolean): Int = 0
+
+    // this is the same weight of a file, allows us to group them together
+    override fun getWeight(): Int = 20
```

Comment:
What is the specific reason for `20`? Seems like a magic number.

Reply:
That's because it is, the value needs to be the same as `PsiFileNode` so that we're able to group them, `PsiFileNode` assigns a static weight of `20` so we mirror that.

Reply:
Is it possible to directly use that weight? If not could you set a comment?

Reply:
Sadly no, because `getWeight` is an instance method, and we inherit from `PsiDirectoryNode` and not `PsiFileNode`. I will add a comment clarifying this.

Reply:
Ah, that's unfortunate. I guess it's not possible to call `PsiDirectoryNode::getWeight()`?

Reply:
Nope, not in Java land I am afraid ._.